core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpServer.java

```diff
@@ -0,0 +1,150 @@
+import io.questdb.WorkerPoolAwareConfiguration;
+import io.questdb.WorkerPoolAwareConfiguration.ServerFactory;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.EagerThreadSetup;
+import io.questdb.mp.SynchronizedJob;
+import io.questdb.mp.WorkerPool;
+import io.questdb.network.IOContextFactory;
+import io.questdb.network.IODispatcher;
+import io.questdb.network.IODispatchers;
+import io.questdb.network.IORequestProcessor;
+import io.questdb.std.Misc;
+import io.questdb.std.ObjList;
+import io.questdb.std.ThreadLocal;
+import io.questdb.std.WeakObjectPool;
+
+public class LineTcpServer implements Closeable {
+    private static final Log LOG = LogFactory.getLog(LineTcpServer.class);
+
+    @Nullable
+    public static LineTcpServer create(
+            CairoConfiguration cairoConfiguration,
+            LineTcpReceiverConfiguration lineConfiguration,
+            WorkerPool sharedWorkerPool,
+            Log log,
+            CairoEngine cairoEngine,
+            MessageBus messageBus
+    ) {
+        if (!lineConfiguration.isEnabled()) {
+            return null;
+        }
+
+        ServerFactory<LineTcpServer, WorkerPoolAwareConfiguration> factory = (netWorkerPoolConfiguration, engine, workerPool, local, bus,
+                functionfactory) -> new LineTcpServer(
+                        cairoConfiguration,
+                        lineConfiguration,
+                        cairoEngine,
+                        workerPool,
+                        bus);
+        LineTcpServer server = WorkerPoolAwareConfiguration.create(lineConfiguration.getWorkerPoolConfiguration(), sharedWorkerPool, log, cairoEngine, factory, messageBus, null);
+        return server;
+    }
+
+    private final IODispatcher<LineTcpConnectionContext> dispatcher;
+    private final LineTcpConnectionContextFactory contextFactory;
+    private final LineTcpMeasurementScheduler scheduler;
+    private final ObjList<LineTcpConnectionContext> busyContexts = new ObjList<>();
+
+    public LineTcpServer(
+            CairoConfiguration cairoConfiguration,
+            LineTcpReceiverConfiguration lineConfiguration,
+            CairoEngine engine,
+            WorkerPool workerPool,
+            MessageBus messageBus
+    ) {
+        this.contextFactory = new LineTcpConnectionContextFactory(engine, lineConfiguration, messageBus, workerPool.getWorkerCount());
+        this.dispatcher = IODispatchers.create(
+                lineConfiguration
+                        .getNetDispatcherConfiguration(),
+                contextFactory);
+        workerPool.assign(dispatcher);
+        scheduler = new LineTcpMeasurementScheduler(cairoConfiguration, lineConfiguration, engine, workerPool);
+        final IORequestProcessor<LineTcpConnectionContext> processor = (operation, context) -> {
+            if (context.handleIO()) {
+                busyContexts.add(context);
```

Comment:
When queue is full, we should put backpressure on TCP stack. Right now we will start filling up `busyContexts` list, which does not feel right.

I would like to suggest cache last context we could not put on the queue and stay on it until it goes away. Only then call dispatcher.processIOQueue()

Reply:
We do need to maintain a list, because once the context cant get a queue event it no longer registers with the despatcher. However, I have changed the code to not call the dispatcher.processIOQueue() until there queue is not blocked.